oommfc/drivers/driver.py

```diff
@@ -156,64 +150,266 @@ def drive(
         else:
-            subdir = f"compute-{system.compute_number}"
+            header = schedule_header
+
+        with oc.util.changedir(workingdir):
+            self.write_mif(
+                system=system,
+                ovf_format=ovf_format,
+                fixed_subregions=fixed_subregions,
+                output_step=output_step,
+                compute=None,
+                **kwargs,
+            )
 
-        workingdir = os.path.join(dirname, system.name, subdir)
+            if runner is None:
+                runner = oc.runner.runner
+            run_cmd = runner._call(
+                argstr=self._miffilename(system), n_threads=n_threads, dry_run=True
+            )
+            with open(schedule_file_name, "wt") as f:
+                f.write(header)
+                f.write("\n")
+                f.write(" ".join(run_cmd))
+
+            stdout = stderr = sp.PIPE
+            if sys.platform == "win32":
+                stdout = stderr = None  # pragma: no cover
+
+            if verbose >= 1:
+                print(
+                    f"Running '{schedule_cmd} {schedule_file_name}' in"
+                    f" '{workingdir.absolute()}'."
+                )
+            system.drive_number += 1
+            res = sp.run(
+                [schedule_cmd, schedule_file_name], stdout=stdout, stderr=stderr
+            )
 
-        # Make a directory inside which OOMMF will be run.
-        if not os.path.exists(workingdir):
-            os.makedirs(workingdir)
+            # remove information about fixed cells for subsequent runs
+            if hasattr(self.evolver, "fixed_spins"):
+                del self.evolver.fixed_spins
+
+            if res.returncode != 0:
+                msg = "Error during job schedule.\n"
+                msg += f"command: {schedule_cmd} {schedule_file_name}\n"
+                if sys.platform != "win32":
+                    # Only on Linux and MacOS - on Windows we do not get stderr and
+                    # stdout.
+                    stderr = res.stderr.decode("utf-8", "replace")
+                    stdout = res.stdout.decode("utf-8", "replace")
+                    msg += f"stdout: {stdout}\n"
+                    msg += f"stderr: {stderr}\n"
+                raise RuntimeError(msg)
+
+    @staticmethod
+    def _setup_working_directory(system, dirname, mode, append=True):
+        system_dir = pathlib.Path(dirname, system.name)
+        try:
+            last_existing_simulation = max(
+                system_dir.glob(f"{mode}*"), key=lambda p: int(p.name.split("-")[1])
+            )
+            next_number = int(last_existing_simulation.name.split("-")[1]) + 1
+        except ValueError:  # glob did not find any directories
+            next_number = 0
+        if next_number > 0 and not append:
+            raise FileExistsError(
+                f"Directory {system.name=} already exists. To "
+                "append drives to it, pass append=True."
+            )
```

Comment:
@marijanbeg The error message (was there already) says that the directory `system.name` (without the `/drive-XXX` or `/compute-XXX`) does already exist and we don't add to it when `append=False`. However, the old (and new) code only tests if there are already sub-directories of the type of simulation we are currently doing (only `drive-XXX` if we use `driver.drive` or only `compute-XXX` if we use `oc.compute`).

What is the desired behaviour here? Do we want to allow writing a compute (drive) run to a system directory that already has a drive (compute) run if `append=False`? Or do we want to make sure that the base directory `system.name` does not exist yet?